src/main/java/com/microsoft/azure/spring/data/documentdb/common/PropertyLoader.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.data.documentdb.common;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+public class PropertyLoader {
+    private static final String PROJECT_PROPERTY_FILE = "/META-INF/project.properties";
+
+    public static String getProjectVersion() {
+        String version = "unknown";
+        InputStream inputStream = null;
+        try {
+            inputStream = PropertyLoader.class.getResourceAsStream(PROJECT_PROPERTY_FILE);
+            if (inputStream != null) {
+                final Properties properties = new Properties();
+                properties.load(inputStream);
+
+                version = properties.getProperty("project.version");
+            }
+        } catch (IOException e) {
+            // Omitted
+        } finally {
+            if (inputStream != null) {
+                try {
+                    inputStream.close();
+                } catch (IOException e) {
```

Comment:
Same as above. If I'd go with if (inputStream != null) {
                                               inputStream.close();
                                            }
Any issue with it?

Reply:
`inputstream.close()` will throw IOException which is checked exception and must be caught here.

Reply:
I get that. I am wondering any practices of handling it so the code is more robust rather than swallow exception. Shall we just re-throw it?

Reply:
Even though we throw the exception here, still the caller has to check the re-thrown exception and deal with the exception, for this user agent case, I think swallow should be enough.

Reply:
ok